js-contribution-guidelines.md

```diff
@@ -0,0 +1,12 @@
+# JS Contribution Guidelines
+
+Many IPFS projects use javascript. Please check these guidelines before contributing js code to an IPFS repository.
+
+## Suggestions
+
+- Build system: Use `npm run` for small projects; for larger ones, use `gulp`.
+- Testing node: `mocha`
+- Testing browser: `karma` + `mocha`
+- Browser building: `webpack` or `browserify` (this fight is still ongoing ;)
```

Comment:
So, confession: I'm a browserify person. I'm not going to suggest one over the other. @diasdavid is right about node devs. I'll remove the fight comment.